cdap-app-fabric/src/main/java/io/cdap/cdap/internal/accelerator/AcceleratorManager.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.accelerator;
+
+import com.google.inject.Inject;
+import io.cdap.cdap.api.metadata.MetadataEntity;
+import io.cdap.cdap.api.metadata.MetadataScope;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.metadata.MetadataAdmin;
+import io.cdap.cdap.proto.id.ApplicationId;
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.cdap.spi.metadata.MetadataRecord;
+import io.cdap.cdap.spi.metadata.ScopedName;
+import io.cdap.cdap.spi.metadata.SearchRequest;
+import io.cdap.cdap.spi.metadata.SearchResponse;
+import org.apache.twill.filesystem.LocationFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Class with helpful methods for dynamic accelerator framework
+ */
+public class AcceleratorManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AcceleratorManager.class);
+  public static final String ACCELERATOR_TAG = "accelerator:%s";
+  public static final String APPLICATION_TAG = "application:%s";
+  public static final String ACCELERATOR_KEY = "accelerator";
+  private final MetadataAdmin metadataAdmin;
+  private final CConfiguration cConf;
+  private static LocationFactory locationFactory;
+
+  @Inject
+  AcceleratorManager(MetadataAdmin metadataAdmin, CConfiguration cConf) {
+    this.metadataAdmin = metadataAdmin;
+    this.cConf = cConf;
+  }
+
+  /**
+   * Returns the list of applications that are having metadata tagged with the accelerator
+   *
+   * @param namespace
+   * @param accelerator
+   * @return
+   * @throws Exception
+   */
+  public Set<ApplicationId> getAppsForAccelerator(NamespaceId namespace,
+                                                  String accelerator) throws Exception {
+    String acceleratorTag = String.format(ACCELERATOR_TAG, accelerator);
+    SearchRequest searchRequest = SearchRequest.of(acceleratorTag).addNamespace(namespace.getNamespace())
```

Comment:
We should support pagination since the number of applications belonged to an accelerator can be larger.

Reply:
Done